main.go

```diff
@@ -282,3 +289,17 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func mapCRDAvailability(ctx context.Context, cl apiclient.Client) (map[string]bool, error) {
+	crdExist := map[string]bool{}
+	crdList := &metav1.PartialObjectMetadataList{}
+	crdList.SetGroupVersionKind(extv1.SchemeGroupVersion.WithKind("CustomResourceDefinitionList"))
+	if err := cl.List(ctx, crdList); apiclient.IgnoreNotFound(err) != nil {
```

Comment:
A list operation will never return a NotFound error
```suggestion
	if err := cl.List(ctx, crdList); err != nil {
```